frontend/src/metabase/visualizations/components/TableInteractive/TableInteractive.jsx

```diff
@@ -563,12 +580,16 @@ class TableInteractive extends Component {
 
     const isLink = cellData && cellData.type === ExternalLink;
     const isClickable = !isLink && !isScrolling;
-    const backgroundColor = this.getCellBackgroundColor(
-      settings,
-      value,
-      rowIndex,
-      column.name,
-    );
+
+    const isIDColumn = value != null && isID(column);
+
+    // Theme options from embedding SDK.
+    const tableTheme = theme?.other?.table;
+
+    const backgroundColor =
+      this.getCellBackgroundColor(settings, value, rowIndex, column.name) ||
+      tableTheme?.cell?.backgroundColor ||
+      "white";
```

Comment:
The `getCellDataTheme` is mainly there to handle the Column ID cases for the cell data. I don't think we can use it to render cells, as there's a difference between cell (outer) and cell data (inner).

That brings up a good point though: `c={tableTheme?.cell?.textColor || "text-brand"}` is not needed in the outer cell, because the inner "cell data" already handles it.

![CleanShot 2567-05-13 at 22 14 04@2x](https://github.com/metabase/metabase/assets/4714175/a3b5db70-9119-4456-bfc6-5c8639d36a52)